### DISCUSSION TITLE
repeat_last_motion (Alt+.) with find_next_char (f) does not respect context

### DISCUSSION THREAD

**Discussion by Alexis-Lapierre** [helix-editor/helix#7710] [General] [answered]
### Summary

For helix 23.05, using repeat_last_motion with find_next_char or till_next_char always extended selection.

However this was deemed 'odd' and was fixed in  #7159

The behaviour on master (using 23.05-245-g505213d4) is now using "f" will respect the context.
In:
 * normal mode => invoke `find_next_char`
 * extend mode => invoke `extend_next_char`
 
 
 This however have the side effect of storing the "extended-ness" the next time we use Alt-.
 This mean that if we type "f" in extended mode to look for something, and then exit back into normal mode, the next Alt-. is till going to extend the selection.
The opposite is also true, typing "f" in normal mode then going into extended mode is going to cancel selection at each Alt-.
 

### Reproduction Steps

 

On helix 23.05-245-g505213d4

I tried this:

1. `hx some_file_with_text`
2.  `fCvj<Alt-.>`

Here is asciinema of input:

[![asciicast](https://asciinema.org/a/oI3KyP38KRbO30m0TDED51DV1.svg)](https://asciinema.org/a/oI3KyP38KRbO30m0TDED51DV1)

I expected this to happen:

Alt-. to extend the selection I made in visual mode

Instead, this happened:

Alt-. cancelled my selection

### Helix log

N/A

### Platform

Linux - Ubuntu 23.04

### Terminal Emulator

kitty

### Helix Version

23.05-245-g505213d4

**Comment by Alexis-Lapierre** [comment]
One easy solution would be to instead having

`find_next_char` and `extend_next_char` in helix-term/src/command.rs
be hard-coded extend / not extend

have the underlying function, `find_next_char`, ask the Context it already have at each call to see if it should extend or not the selection depending on something like `Context.editor.mode == Mode::Select`

**Comment by pascalkuthe** [comment, marked as answer]
This is on purpose. The concept of visual mode as a special mode will be removed entirely in the future. Visual mode will just be a keymap mode, with each command having a an extend and non-extend variant

**Reply by Alexis-Lapierre** [reply]
OK thanks for the response.

For people that really like the old behaviour, it is pretty easy to keep it with 5 lines to add in config.toml:

```toml
[keys.normal]
f = "extend_next_char"
F = "extend_prev_char"
t = "extend_till_char"
T = "extend_till_prev_char"
```
